tools/linker/RemoveUserResourcesSubStep.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Mono.Cecil;
+using Mono.Linker;
+using Mono.Tuner;
+using MonoTouch.Tuner;
+
+namespace Xamarin.Linker {
+
+	public class RemoveUserResourcesSubStep : ExceptionalSubStep {
+
+		public RemoveUserResourcesSubStep (LinkerOptions options)
+		{
+			Device = options.Device;
+		}
+
+		public override SubStepTargets Targets {
+			get { return SubStepTargets.Assembly; }
+		}
+
+		public bool Device { get; private set; }
+
+		protected override string Name { get; } = " Removing User Resources";
+		protected override int ErrorCode { get; } = 2030;
+
+		protected override void Process (AssemblyDefinition assembly)
+		{
+			if (Profile.IsProductAssembly (assembly) || Profile.IsSdkAssembly (assembly))
+				return;
+
+			var module = assembly.MainModule;
+			if (!module.HasResources)
+				return;
+			
+			HashSet<string> libraries = null;
+			if (assembly.HasCustomAttributes) {
+				foreach (var ca in assembly.CustomAttributes) {
+					if (!ca.AttributeType.Is ("ObjCRuntime", "LinkWithAttribute"))
+						continue;
+					var lwa = Xamarin.Bundler.Assembly.GetLinkWithAttribute (ca);
+					if (libraries == null)
+						libraries = new HashSet<string> (StringComparer.OrdinalIgnoreCase);
+					libraries.Add (lwa.LibraryName);
+				}
+			}
+
+			if (libraries == null)
+				return;
```

Comment:
Assemblies can have MonoTouch resources (`__monotouch_content_` / `__monotouch_page_ `) without having `LinkWith` attributes, so I don't think this check is correct.